### PR TITLE
setHovered to false onClick

### PR DESCRIPTION
**Pull request by daniel-covelli**
The aim of this PR is to prevent overlays from lingering after the onClick handler is triggered. The current unwanted behavior can be seen below.

c177a5a5d36c81b654d35f2af95841a7

This change will set isHovered to false once a click is registered on the overlay.

### CODE REVIEW

**Comment by giannif** [comment]
@daniel-covelli a lot of the time we don't want the overlay to disappear when you click, visit giphy.com on desktop and see how the overlays have buttons you click to copy links, etc. 

I'm wondering why isHovered interferes with the UI you have pictured above? Can you tell me more about the problem you're trying to solve?

**Comment by daniel-covelli** [comment]
@giannif so the Giphy Grid component above is wrapped in a [Ant Design Popover](https://ant.design/components/popover/#header) component which I am a opening and closing via a state change. After selecting a gif in the grid component, I am closing the visibility state of the popover back to _false_, preventing the onMouseLeave method, in the Gif component, from being called.  After the gif has been processed by our backend and displayed in the frontend, if a user opens the same overlay again, the isHovered state for the selected gif is still set to true. 

Basically I am trying to find another way to set isHovered to false without the onMouseLeave method.

**Comment by giannif** [comment]
@daniel-covelli gotcha, I'm thinking maybe we can set isHovered to false when the intersection observer fires false, but I'm trying to reproduce your scenario and can't, maybe it's browser-specific? 

Here's my attempt:
https://codesandbox.io/s/ishovered-check-rvcgi?file=/src/index.tsx

**Comment by daniel-covelli** [comment]
@giannif In the code you provided, you are wrapping your Grid in a div that changes opacity based on the "show/hide" button. In my case, I am wrapping the Grid in, what is basically, a modal, which opens and closes. My guess is that once I close the modal, the Gif component can not register where the mouse is because its parent div has been removed from the DOM.  In your case, the location of the mouse can still be registered by each Gif component, despite it not being visible.

**Comment by giannif** [comment]
@daniel-covelli yeah I tried removing from the dom, same behavior. I'm not sure what's happening differently in your environment, feel free to fork the code sandbox to get it to the state where the bug is reproducible

**Comment by daniel-covelli** [comment]
@giannif hmm weird... I am also using the SearchContextManager, maybe that has something to do with it. `import { useState, useEffect, useRef } from 'react'

import {
  Grid,
  SearchContextManager,
  GifOverlayProps
} from '@giphy/react-components'
import { Popover, Input, Button } from 'antd'

import { GiphyFetch } from '@giphy/js-fetch-api'
import { getStaticRoute } from '@website/utils'
import { SearchOutlined, CloseCircleFilled } from '@ant-design/icons'

// called by CommentUploader.tsx
const GiphyCommentSelector = ({ onSelectGif }) => {
  // determines giphy selector state
  const [visible, setVisible] = useState(false)
  const giphySelector = (
    <SearchContextManager apiKey={process.env.GIPHY_API_KEY}>
      <Components onSelectGif={onSelectGif} setVisible={setVisible} />
    </SearchContextManager>
  )

  return (
    <Popover
      content={giphySelector}
      trigger={'click'}
      visible={visible}
      onVisibleChange={() => {
        setVisible(!visible)
      }}
    >
      <Button className={'giphyPopover__btn'}>GIF</Button>
    </Popover>
  )
}

const Components = ({ onSelectGif, setVisible }) => {
  const [query, setQuery] = useState('')
  const overlayRef = useRef(null)
  const [hovered, setHovered] = useState(false)
  // giphy instance
  const giphyFetch = new GiphyFetch(process.env.GIPHY_API_KEY)
  // request for trending gifs
  //   const onOpenGifs = (offset: number) =>
  //     giphyFetch.search('meme', { offset, limit: 12 })
  // request for
  const onOpenGifs = (offset: number) =>
    giphyFetch.trending({ offset, limit: 10 })
  // search request for gifs
  const fetchGifs = (offset: number) =>
    giphyFetch.search(`${query}`, { offset, limit: 12 })
  // unique react key per response
  const searchKey = !query ? 'gifs' : `${query} / gifs`
  // if no query, return trendingGifs
  const currentReq = searchKey === 'gifs' ? onOpenGifs : fetchGifs
  // grid item overlay
  const Overlay = ({ gif, isHovered }: GifOverlayProps) => {
    return isHovered ? <div ref={overlayRef} className="overlay"></div> : null
  }


  return (
    <>
      <Input
        value={query}
        onChange={e => setQuery(e.target.value)}
        placeholder="search GIPHY"
        style={{ marginTop: '7px' }}
        suffix={
          query === '' ? (
            <SearchOutlined style={{ color: ' #81869b' }} />
          ) : (
            <a onClick={() => setQuery('')}>
              <CloseCircleFilled style={{ color: ' #81869b' }} />
            </a>
          )
        }
      />
      <div className={'giphyGrid_div'}>
        <Grid
          key={searchKey}
          columns={3}
          width={350}
          fetchGifs={currentReq}
          noLink={true}
          overlay={Overlay}
          onGifClick={(gif, e) => {
            console.log('GIF: ', gif)
            onSelectGif(gif)

            setQuery('')
            setVisible(false)
          }}
        />
      </div>
    </>
  )
}

export default GiphyCommentSelector
`

**Comment by daniel-covelli** [comment]
@giannif I cant reproduce the bug, I've tried for the past 40 ish minutes. The only difference now is that I am passing the equivalent of the App component into a Popover antd component. 

Here is my attempt: https://codesandbox.io/s/ishovered-check-forked-2oft1?file=/src/index.tsx

If you have any thoughts as to what might be happening, please let me know.

**Comment by daniel-covelli** [comment]
@giannif got it to work using the above logic. seems that the bug had something to do with how the antd popover component renders its content. Thanks for you help tho!

**Comment by giannif** [comment]
@daniel-covelli no problem, glad you got it fixed! 👍
